Add Nokia 8110, 6300 and 8000 4G (argon, leo, sparkler)

affenull2345 opened the pull request:
Add support for the Nokia 8110 4G (argon), Nokia 6300 4G (leo) and Nokia 8000 4G (sparkler).

Also add required PWM, vibrator, charger and battery nodes in qcom-pm8909.dtsi ~~and a new file qcom-msm8905.dtsi, which includes qcom-msm8909.dtsi and removes two CPU nodes because MSM8905 is just MSM8909 with two CPUs instead of four~~ (CPU removal is now handled in https://github.com/msm8916-mainline/lk2nd/pull/214).

TravMurav commented on arch/arm/boot/dts/qcom-msm8905-nokia-argon.dts:
```diff
@@ -0,0 +1,407 @@
+		compatible = "nokia,argon-gc9305-v2-panel", "panel-mipi-dbi-spi";
+		reg = <0>;
+		backlight = <&backlight>;
+		//te-gpios = <&tlmm 24 GPIO_ACTIVE_HIGH>;
```
Should probably drop comments like this across the node

TravMurav commented on arch/arm/boot/dts/qcom-msm8905.dtsi:
```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: GPL-2.0-only
```
Please separate commits: Introduce SoC dtsi then introduce devices one-by-one (can probably combine ones that share a dtsi)

TravMurav commented on arch/arm/boot/dts/qcom-msm8909-nokia-leo-common.dtsi:
```diff
@@ -0,0 +1,367 @@
+		compatible = "panel-mipi-dbi-spi";
+		reg = <0>;
+		backlight = <&backlight>;
+		//te-gpios = <&tlmm 24 GPIO_ACTIVE_HIGH>;
```
ditto

TravMurav commented on arch/arm/boot/dts/qcom-pm8909.dtsi:
```diff
@@ -39,6 +39,13 @@
 			};
 		};
 
+		pm8909_charger: charger@1000 {
```
Please have separate commits for not-yet-upstreamed drivers like lbc and vm-bms:

- add soc dtsi
- add device foo
- add lbc and vm-bms to pmic dtsi
- device foo: enable lbc and bms

This way upstreamable changes will be ready in a good shape instead of having dependency on not-yet upstreamed thing that may change later

TravMurav commented on arch/arm/boot/dts/qcom-msm8909-nokia-leo-common.dtsi:
```diff
@@ -0,0 +1,367 @@
+			&tlmm 6 0
+			&tlmm 1 0>;
+
+		/* Specify linux,keypad in device dts */
```
Maybe you want status=disabled here so the incomplete node doesn't get used in case some other device includes this dtsi

TravMurav commented on arch/arm/boot/dts/qcom-msm8905-nokia-argon.dts:
```diff
@@ -0,0 +1,367 @@
+#include <dt-bindings/pinctrl/qcom,pmic-mpp.h>
+
+/ {
+	model = "Nokia 8110 4G (argon)";
```
nit: I think having a codename here is redundant

TravMurav commented on arch/arm/boot/dts/qcom-msm8905-nokia-argon.dts:
```diff
@@ -0,0 +1,367 @@
+
+/ {
+	model = "Nokia 8110 4G (argon)";
+	compatible = "nokia,argon", "qcom,msm8909";
```
nit: Probably `qcom,msm8905`?

affenull2345 replied:
Maybe it's better to have both? MSM8905 is almost fully compatible with MSM8909

stephan-gh replied:
I'm still wondering if we should fixup the "partial goods" related code to work properly for mainline, and then avoid the `msm8905.dtsi` completely. It looks like 8905 can theoretically have 2 or 4 cores (although I doubt any feature phone got the 4 core version).

There is some register that should be checked if there are 2 or 4 cores available: https://github.com/msm8916-mainline/lk2nd/commit/a4b4c47f068995ade5b8267ec38b7b76d9e4d769

affenull2345 replied:
I'm going to remove the msm8905 include for now and work on fixing partial_goods in lk2nd.

TravMurav commented on arch/arm/boot/dts/qcom-msm8905-nokia-argon.dts:
```diff
@@ -0,0 +1,367 @@
+		key-back {
+			wakeup-source;
+			gpios = <&tlmm 98 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_BACKSPACE>;
```
nit: Unsure if this phone's keyboard has a key that semantically matches "backspace" :D

But it's pretty funny to pick those keys right now as I believe there is no "usual" software for those phones to tailor the keymap for?

affenull2345 replied:
The "end call"/"back" key on the 8110 doubles as a backspace key, so I think it makes sense to call it backspace. KaiOS also calls it backspace. The 8000 and 6300 have a separate "back" key (backspace) and a separate "end call" key (esc because it closes applications).

I assigned the left soft key to tab for use with terminals. The other keys are freely chosen. The keycodes don't really matter because the software can remap the keys anyway, so I'll change them to make more sense.

TravMurav reviewed (approved):
Some small nitpicks, otherwise looks good to me

stephan-gh commented on arch/arm/boot/dts/qcom-msm8905-nokia-argon.dts:
```diff
@@ -0,0 +1,361 @@
+		default-brightness-level = <10>;
+	};
+
+	gpio_keys {
```
Avoid underscores in node names

stephan-gh commented on arch/arm/boot/dts/qcom-msm8905-nokia-argon.dts:
```diff
@@ -0,0 +1,361 @@
+		};
+	};
+
+	matrix_keypad: matrix_keypad@0 {
```
Maybe just `keypad {`? Avoid underscores, also the `@0` shouldn't be here since the child has no `reg`.

stephan-gh commented on arch/arm/boot/dts/qcom-msm8905-nokia-argon.dts:
```diff
@@ -0,0 +1,361 @@
+};
+
+&tlmm {
+	gpio_key_active: gpio_key_active {
```
`gpio-key-active-state` and so on

stephan-gh commented on arch/arm/boot/dts/qcom-msm8905-nokia-argon.dts:
```diff
@@ -0,0 +1,361 @@
+
+&tlmm {
+	gpio_key_active: gpio_key_active {
+		pins = "gpio90", "gpio98";
```
no function?

stephan-gh commented on arch/arm/boot/dts/qcom-msm8905-nokia-argon.dts:
```diff
@@ -0,0 +1,361 @@
+
+		drive-strength = <2>;
+		bias-disable;
+		output-low;
```
Would expect the keypad gpios to be input, not output

affenull2345 replied:
It's a matrix keypad, either the col-gpios or the row-gpios have to be output. It's wired up like this:

```
< - O - O - O
    |   |   |
< - O - O - O
    |   |   |
< - O - O - O
    |   |   |
< - O - O - O
    |   |   |
< - O - O - O
    |   |   |
    ^   ^   ^
```
(`O`=key, `^`=output gpio, `<`=input gpio)

stephan-gh commented on arch/arm/boot/dts/qcom-msm8905-nokia-argon.dts:
```diff
@@ -0,0 +1,361 @@
+			hfront-porch = <0>;
+			vfront-porch = <0>;
+			hsync-len = <0>;
+			vsync-len = <0>;
```
Why are all of these 0? Is this the same on downstream? (To be fair, even today I still don't fully understand what all those porch values do)

affenull2345 replied:
They aren't 0 on downstream, but the mainline driver doesn't fully understand them and the display works fine (except for tearing issues, but those are probably because frame synchronization is not supported)

stephan-gh replied:
What does "the mainline driver doesn't fully understand them" mean exactly? Does it look bad if you set them like on downstream?

affenull2345 replied:
It fails with "panel-timing out of bounds" because it expects only back-porch to be set (see https://github.com/msm8916-mainline/linux/tree/master/drivers/gpu/drm/tiny/panel-mipi-dbi.c#L252)

affenull2345 replied:
And setting the porch values would change the frame size sent to the display, which currently matches the size sent on downstream, and nothing besides that, so I think it's good they're set to zero

stephan-gh replied:
Thanks, let's leave it as-is for now then.

stephan-gh commented on arch/arm/boot/dts/qcom-msm8905-nokia-argon.dts:
```diff
@@ -0,0 +1,361 @@
+
+&usb {
+	status = "okay";
+	extcon = <&pm8909_usbin>;
```
You need to make pm8909_usbin `status = "okay"` (and this should be removed in the LBC commits because then it handles the extcon and interrupt)

stephan-gh reviewed (commented):
This looks very good already, thanks! Just some quick comments for now, will need to take a closer look at the stuff I'm not familiar with yet (mainly the matrix-keypad and the panel stuff)

stephan-gh commented on arch/arm/boot/dts/qcom-msm8905-nokia-argon.dts:
```diff
@@ -0,0 +1,397 @@
+
+			0x0300001C      /* (0,3) KEY_ENTER */
+			0x0301006A      /* (1,3) KEY_RIGHT */
+			0x03020111      /* (3,3) BTN_RIGHT */
```
Can you switch this to use the MATRIX_KEY macro so it is properly readable? https://elixir.bootlin.com/linux/v6.1-rc4/source/include/dt-bindings/input/input.h#L15

There are examples for this in various DTs.

stephan-gh commented on arch/arm/boot/dts/qcom-msm8905-nokia-argon.dts:
```diff
@@ -0,0 +1,398 @@
+			vfront-porch = <0>;
+			hsync-len = <0>;
+			vsync-len = <0>;
+			clock-frequency = <2304000>; /* for 30 FPS */
```
`<(240 * 320 * 30)>; /* 30 fps */` might be a bit clearer

stephan-gh commented on arch/arm/boot/dts/qcom-msm8905-nokia-argon.dts:
```diff
@@ -0,0 +1,398 @@
+	status = "okay";
+};
+
+&blsp_spi5_default {
```
I know you have just taken these overrides from downstream but I'd still like to understand them a bit better:

1. The cs-gpio is usually assigned the `gpio` function on mainline (the default in `qcom-msm8909.dtsi` because controlling it through the SPI controller usually has unnecessary overhead. Also, since only one device (the panel) is connected to the SPI bus there is no need to ever change its state at all. I'd expect that it just works when dropping the cs part here.
2. Is the weird `bias-pull-up` really necessary to make it work? What if you remove the `&blsp_spi5_default` and `&blsp_spi5_sleep` overrides completely, does it still work?

affenull2345 replied:
The display stayed white until I set the cs pin function to `blsp_spi5`, and it took me a few days of trial-and-error to figure that out. ~~But now I realized that using `output-low` instead of `output-high` also fixes the issue, so I'll use that instead.~~ The pull-up values are from downstream and don't seem to affect the display behavior on the 8110, which means they can be removed. (What made the downstream developers use those weird pull-up values?)

stephan-gh replied:
Oops I thought it's active-high but it looks like it is active-low:

> The CS line is normally held high, which disconnects the peripheral from the SPI bus. (This type of logic is known as “active low,” and you’ll often see used it for enable and reset lines.) Just before data is sent to the peripheral, the line is brought low, which activates the peripheral.

https://learn.sparkfun.com/tutorials/serial-peripheral-interface-spi/all#chip-select-cs

I'll need to re-check the default in `qcom-msm8909.dtsi` then...

affenull2345 replied:
I'm afraid we'll have to stick with the `blsp_spi5` function for the cs pin, otherwise I get a white screen after turning it off and back on.

stephan-gh replied:
Looking closer I think it is not supposed to be used with a static pinctrl set to `output-low`. There are two options:

1. Use CS pin control of the SPI controller. A closer look suggests that the overhead I mentioned [only affects newer SoCs](https://lore.kernel.org/linux-arm-msm/20200624170746.1.I997a428f58ef9d48b37a27a028360f34e66c00ec@changeid/) (geni SPI instead of spi-qup). For spi-qup the CS pin control was broken on older SoCs but [at some point re-enabled for newer ones](https://lore.kernel.org/linux-arm-msm/1501224781-19519-2-git-send-email-varada@codeaurora.org/) (spi-qup-v2) such as MSM8909. In other words, it should be fine to make use of it here.
2. Control CS pin using GPIOs. The part we missed is that this also requires adding `cs-gpios` to the SPI controller to make it set that high/low as needed. See the [diff in this patch](https://lore.kernel.org/linux-arm-msm/20200624170746.1.I997a428f58ef9d48b37a27a028360f34e66c00ec@changeid/).

Both approaches should work fine, but since I cannot think of any disadvantage for using (1) we should probably just go with using the CS pin control of the SPI controller. We should probably change this in `qcom-msm8909.dtsi` and drop the `output-high` there.

affenull2345 replied:
Does the CS pin still need a separate node with a different drive strength (as it has now) or does that make no difference?

stephan-gh replied:
No idea, I guess it doesn't matter that much so we might as well combine them I guess.

stephan-gh commented on arch/arm/boot/dts/qcom-msm8905-nokia-argon.dts:
```diff
@@ -0,0 +1,362 @@
+	};
+};
+
+&blsp_uart1 {
```
Please order this alphabetically, or alternatively, according to https://wiki.postmarketos.org/wiki/MSM8916_Mainlining#Device_Tree_Structure

stephan-gh commented on arch/arm/boot/dts/qcom-msm8905-nokia-argon.dts:
```diff
@@ -0,0 +1,362 @@
+		pinctrl-names = "default";
+		pinctrl-0 = <&matrix_keypad_row_default &matrix_keypad_col_default>;
+
+		row-gpios = <&tlmm 97 0
```
I'd replace all the 0 with GPIO_ACTIVE_LOW (even when it's ignored because the gpio-activelow counts atm)

affenull2345 replied:
I'm not sure I really understand this. I have two questions:

1. Why do you say GPIO_ACTIVE_LOW is ignored when `gpio-activelow` is set? Looking at the drivers, it seems that `gpio-activelow` just makes matrix-keypad invert the gpio values.
2. What is the preferred polarity when it doesn't really matter? The column polarity must match the row polarity, but it can be either active-low or active-high, and this shouldn't change the behavior of the keypad. Downstream uses active-low.

I've removed the `gpio-activelow` because it seems that `GPIO_ACTIVE_LOW` is the preferred method to set the polarity.

stephan-gh replied:
The `gpiod_*` APIs in Linux handle the polarity internally, based on the `GPIO_ACTIVE_HIGH`/`GPIO_ACTIVE_LOW` you specify in the device tree. However, the matrix-keypad driver is still using the old `gpio_*` APIs. Those do not respect the polarity. Adding `GPIO_ACTIVE_LOW` would just be "future-proofing" for when the matrix-keypad driver is updated to use the newer APIs. Until then, only `gpio-activelow` should have an effect.

I can't comment on whether you should use active-high or active-low right now, guess I'll need to go back to actually understanding how the matrix-keypad works :sweat_smile:

stephan-gh replied:
> What is the preferred polarity when it doesn't really matter? The column polarity must match the row polarity, but it can be either active-low or active-high, and this shouldn't change the behavior of the keypad. Downstream uses active-low.

I think you're right that it's up to us here. But I think this is not always the case, AFAIU the bias/pull of the row GPIOs needs to match the polarity of the matrix keypad. We can change the row GPIO bias/pull using pinctrl here, but on other platforms it could be hardwired using additional resistors.

The bias/pull of the row GPIOs is used when a scan is started. First all col GPIOs are switched to input mode (floating). This means all row GPIOs will take the state of their bias/pull. You have `bias-pull-up` so they become high (inactive with active-low). Then the col GPIOs are driven low one by one and if a button is pressed the row GPIO will turn low (active).

While it probably makes very little difference, I'd probably just follow the original configuration on downstream here. But I still think you need `gpio-activelow;` to get the active-low behavior, even when you use `GPIO_ACTIVE_LOW`.

stephan-gh commented on arch/arm/boot/dts/qcom-msm8905-nokia-argon.dts:
```diff
@@ -0,0 +1,363 @@
+		reset-gpios = <&tlmm 25 GPIO_ACTIVE_HIGH>;
+		dc-gpios = <&tlmm 110 GPIO_ACTIVE_HIGH>;
+		spi-max-frequency = <50000000>;
+		power-supply = <&pm8909_l17>, <&pm8909_l6>;
```
Are you sure this works? I would expect that it only uses the first regulator (you can check in `/sys/kernel/debug/regulator/regulator_summary`, if the display is listed as consumer under both or just l17.

What I don't understand is why the panel-mipi-dbi only has a `power-supply`. The bindings clearly [talk about the separate Vdd and Vddi supply](https://github.com/msm8916-mainline/linux/blob/36a4c82a90cbf1816b66147b2d76ff418d96ce14/Documentation/devicetree/bindings/display/panel/panel-mipi-dbi-spi.yaml#L24-L25) but then only vdd is implemented in the driver. Maybe this should be fixed upstream...

stephan-gh replied:
Ah on the line below it says they were combined into power-supply here, but it's still bad :/

stephan-gh commented on arch/arm/boot/dts/qcom-msm8905-nokia-argon.dts:
```diff
@@ -0,0 +1,363 @@
+
+		drive-strength = <2>;
+		bias-disable;
+		output-low;
```
I think this is unnecessary since the gpio-matrix-keypad driver will initialize those themselves like it wants to.

stephan-gh commented:
Can you update to the `msm8916/6.1-rc5` branch? I added the `qcom-msm8909-pm8909.dtsi` include similar to the `msm8916-pm8916.dtsi` used on all the MSM8916 devices, so please:

- `#include "qcom-msm8909-pm8909.dtsi"` instead of `#include "qcom-msm8909.dtsi"` and `#include "qcom-pm8909.dtsi"`
- Drop all the `-supply`s that are already in `qcom-msm8909-pm8909.dtsi`
- Change the regulator definitions to be like in MSM8916 devices:
  - `&rpm_requests` -> `&smd_rpm_regulators` (drop the extra subnode)
  - Optional: Can drop the label from all regulator nodes (e.g. `pm8909_l1: l1 { ... }` -> `l1 { ... };`
- See also: https://wiki.postmarketos.org/wiki/MSM8916_Mainlining#Regulators

You can also drop the following 3 commits since I squashed/applied them already:
- `ARM: dts: qcom: pm8909: add PWM and vibrator`
- `ARM: dts: qcom: msm8909: control SPI CS pins through the SPI controller`
- `ARM: dts: qcom: pm8909: add LBC and BMS`

I also renamed `&wcnss` -> `&pronto` for consistency with `msm8916.dtsi` (even though I prefer the `&wcnss` name, should change it upstream for everything).

Thanks!

stephan-gh commented:
Also add the `/dts-v1/;` at the top of your DTS files, not sure why I added it in `qcom-msm8909.dtsi`.

stephan-gh commented on arch/arm/boot/dts/qcom-msm8905-nokia-argon.dts:
```diff
@@ -0,0 +1,327 @@
+	reserved-memory {
+		/delete-node/ rmtfs@87c00000;
+		/* On downstream, this address is set by the bootloader */
+		rmtfs@9f700000 {
```
In theory the rmtfs address should be dynamic and not fixed to one particular address, although in practice it's often not. Have you tested that this override is necessary and that the default rmtfs does not work?

affenull2345 replied:
The modem firmware requires a size of 0x300000, which does not fit at the default address because of the other regions.

stephan-gh commented on arch/arm/boot/dts/qcom-msm8905-nokia-argon.dts:
```diff
@@ -0,0 +1,327 @@
+		reg = <0>;
+		backlight = <&backlight>;
+		reset-gpios = <&tlmm 25 GPIO_ACTIVE_HIGH>;
+		dc-gpios = <&tlmm 110 GPIO_ACTIVE_HIGH>;
```
Please add pinctrl for these two (all GPIOs in general) to ensure they are configured correctly no matter what runs before Linux. I think the default is `bias-pull-down` and `drive-strength = <2>` so you could just use that if you're unsure I guess. Alternatively `bias-disable` since it shouldn't really matter for output GPIOs.

affenull2345 replied:
The driver doesn't seem to support pinctrl. I guess it needs another patch upstream.

stephan-gh replied:
The basic pinctrl stuff (`pinctrl-names = "default";`) is implemented in the driver core and shouldn't need special support in the driver.

stephan-gh commented on arch/arm/boot/dts/qcom-msm8909-nokia-leo-common.dtsi:
```diff
@@ -0,0 +1,285 @@
+		/delete-node/ wcnss@8d500000;
+		fih_region@84a00000 {
+			reg = <0x84a00000 0x00400000>;
+			compatible = "removed-dma-pool";
```
I think this is unneeded, all the other nodes survive without this.

stephan-gh commented on arch/arm/boot/dts/qcom-msm8909-nokia-leo-common.dtsi:
```diff
@@ -0,0 +1,285 @@
+		/delete-node/ mpss@88000000;
+		/delete-node/ rmtfs@87c00000;
+		/delete-node/ wcnss@8d500000;
+		fih_region@84a00000 {
```
Node names shouldn't contain underscores, `fih@` would probably be enough. But any idea what this even is? What "fih" stands for?

affenull2345 replied:
As you probably have guessed, this name is used on downstream. AFAIK it stands for Foxconn Mobile Holdings, which is the ODM that manufactures/co-designs/etc these devices. I don't see any reason to keep this node, but I think they wanted Linux to stay out of that memory range because some other part of the firmware uses it. (That's what removed-dma-pool is for)

stephan-gh replied:
It's probably better to keep it to avoid hard-to-debug crashes later on. But the `compatible` should be unneeded (similar to other existing reservations) and maybe call this `fih@` or even just `reserved@`.

TravMurav commented on arch/arm/boot/dts/qcom-msm8905-nokia-argon.dts:
```diff
@@ -0,0 +1,357 @@
+&pm8909_charger {
+	status = "okay";
+
+	qcom,vdd-safe = <4400000>;
+	qcom,ibat-safe = <540000>;
```
We usually drop the charger target 0.05v from downstream just in case (and 4.4 sounds more than usual for a cell tbh, it's usually 4.35 iirc)

Also please set the ibat-safe to the lowest out of (ibat-safe|ibat-max*) in the downstream (360mA afaict) as this driver sets both to the same value

affenull2345 committed:
drm/tiny: panel-mipi-dbi: Read I/O supply from DT

To support platforms with a separate I/O voltage supply, set the new
io_regulator property along with the regulator property of the DBI
device. Read the I/O supply from a new "io-supply" device tree
property.

Signed-off-by: Otto Pflüger <otto.pflueger@abscue.de>

affenull2345 committed:
dt-bindings: display: panel: mipi-dbi-spi: Add io-supply

Add documentation for the new io-supply property, which specifies the
regulator for the I/O voltage supply on platforms where the panel
panel power and I/O supplies are separate.

Signed-off-by: Otto Pflüger <otto.pflueger@abscue.de>

affenull2345 committed:
drm/mipi-dbi: Support separate I/O regulator

The MIPI DBI specification defines separate vdd (panel power) and
vddi (I/O voltage) supplies. Displays that require different voltages
for the different supplies do exist, so the supplies cannot be
combined into one as they are now. Add a new io_regulator property to
the mipi_dbi_dev struct which can be set by the panel driver along
with the regulator property.

Signed-off-by: Otto Pflüger <otto.pflueger@abscue.de>

affenull2345 committed:
ARM: dts: qcom: msm8905-nokia-argon: enable LBC and BMS

Signed-off-by: Otto Pflüger <otto.pflueger@abscue.de>

affenull2345 committed:
ARM: dts: qcom: msm8909-nokia-leo-common: enable LBC and BMS

Signed-off-by: Otto Pflüger <otto.pflueger@abscue.de>

stephan-gh commented on Documentation/devicetree/bindings/display/panel/panel-mipi-dbi-spi.yaml:
```diff
@@ -80,6 +80,9 @@ properties:
       Controller data/command selection (D/CX) in 4-line SPI mode.
       If not set, the controller is in 3-line SPI mode.
 
+  power-supply:
```
FWIW I think this is currently inherited from `panel-common.yaml`.

stephan-gh commented on arch/arm/boot/dts/qcom-msm8905-nokia-argon.dts:
```diff
@@ -0,0 +1,363 @@
+		pinctrl-names = "default";
+		pinctrl-0 = <&keypad_backlight_gpio>;
+
+		compatible = "gpio-leds";
```
Please make compatible the first property for better readability

stephan-gh commented on arch/arm/boot/dts/qcom-msm8905-nokia-argon.dts:
```diff
@@ -0,0 +1,363 @@
+		pins = "gpio1";
+		function = PMIC_GPIO_FUNC_NORMAL;
+
+		output-low;
```
this should have a `power-source` property, looks like you want `qcom,vin-ctrl = <0x03>;`, aka PM8916_GPIO_L5 (1.8V)

Interestingly leo has qcom,vin-ctrl = <0x02>; listed there (aka PM8916_GPIO_L2, 1.15V). It probably doesn't make much of a difference for just controlling that transistor used to enable the backlight...

Also looks like they hacked in some code to configure a `bias-pull-down;` in https://github.com/msm8916-mainline/linux-downstream/blob/502781fd6d10e36a20f653cea1aff71feb24b55a/drivers/leds/leds-qpnp.c#L1090-L1093

stephan-gh commented on arch/arm/boot/dts/qcom-msm8905-nokia-argon.dts:
```diff
@@ -0,0 +1,364 @@
+	};
+
+	l17 {
+		regulator-min-microvolt = <2800000>;
```
looks like this should be also 2.85V based on https://github.com/msm8916-mainline/linux-downstream/blob/502781fd6d10e36a20f653cea1aff71feb24b55a/arch/arm/boot/dts/argon/msm8909-mdss.dtsi#L111-L118 (and similar thing in LEO.dtsi)

affenull2345 committed:
ARM: dts: qcom: add msm8905-nokia-argon (Nokia 8110 4G)

Signed-off-by: Otto Pflüger <otto.pflueger@abscue.de>

affenull2345 committed:
ARM: dts: qcom: add msm8909-nokia-leo and msm8909-nokia-sparkler

Add support for the Nokia 6300 4G (leo) and the Nokia 8000 4G (sparkler).
The devices are very similar, so most of the device tree is shared
between the two devices in qcom-msm8909-nokia-leo-common.dtsi.

Signed-off-by: Otto Pflüger <otto.pflueger@abscue.de>

stephan-gh commented:
In addition to the above, can you rebase to `msm8916/6.1-rc8` and test briefly that it still works fine? I'm planning to merge this next week. Thanks :)

stephan-gh commented:
Integrated into `wip/msm8916/6.1`. Thanks!
